WordPress/Classes/ViewRelated/Reader/Tags View/ReaderTopicCollectionViewCoordinator.swift

```diff
@@ -40,45 +40,45 @@ class ReaderTopicCollectionViewCoordinator: NSObject {
 
     weak var delegate: ReaderTopicCollectionViewCoordinatorDelegate?
 
-    let collectionView: UICollectionView
+    weak var collectionView: UICollectionView?
     var topics: [String] {
         didSet {
             reloadData()
         }
     }
 
-    deinit {
-        guard let layout = collectionView.collectionViewLayout as? ReaderInterestsCollectionViewFlowLayout else {
-            return
-        }
-
-        layout.isExpanded = false
-        layout.invalidateLayout()
-    }
-
     init(collectionView: UICollectionView, topics: [String]) {
         self.collectionView = collectionView
         self.topics = topics
 
         super.init()
 
-        configureCollectionView()
+        configure(collectionView)
+    }
+
+    func invalidate() {
+        guard let layout = collectionView?.collectionViewLayout as? ReaderInterestsCollectionViewFlowLayout else {
+            return
```

Comment:
Just a tangential though. Should we make the app crash in development if the type is not the one expected? And/or log an error in production.

Reply:
I don't think this return line would ever execute though. The three things involved here, the collection view, the layout, and this coordinator are tightly coupled together. I don't think you can simply swap out the layout type during development without realizing something wrong visually.

Also, there is a [fatalError](https://github.com/wordpress-mobile/WordPress-iOS/blob/33fdc51b0532600997bdf97fc92ffc05244b102d/WordPress/Classes/ViewRelated/Reader/Tags%20View/ReaderTopicCollectionViewCoordinator.swift#L176) in this class that guarantees that the layout instance can't be any other type.

Reply:
>  I don't think you can simply swap out the layout type during development without realizing something wrong visually.

Fair enough. 👍